src/utils/hooks/useCreateReview.tsx

```diff
@@ -1,45 +1,67 @@
+import { useGetFullReviewStructureAsync } from './useGetReviewStructureForSection'
 
-interface UseCreateReviewProps {
+// below lines are used to get return type of the function that is returned by useCreateReviewMutation
+type UseCreateReviewMutationReturnType = ReturnType<typeof useCreateReviewMutation>
+type PromiseReturnType = ReturnType<UseCreateReviewMutationReturnType[0]>
+// hook used to start a review, , as per type definition below (returns promise that resolve with mutation result data)
+type UseCreateReview = (props: {
   structure: FullStructure
   assignment: AssignmentDetails
-}
+}) => () => PromiseReturnType
+
+type ConstructReviewInput = (structure: FullStructure) => ReviewInput
 
-const useCreateReview = ({ assignment }: UseCreateReviewProps) => {
-  const [error, setError] = useState<ApolloError | undefined>()
+const useCreateReview: UseCreateReview = ({ structure, assignment }) => {
+  const [updateReview] = useCreateReviewMutation()
 
-  const [createReviewMutation] = useCreateReviewMutation({
-    onError: (error) => setError(error),
+  const getFullReviewStructureAsync = useGetFullReviewStructureAsync({
+    fullApplicationStructure: structure,
+    reviewAssignment: assignment,
   })
 
-  const createReviewFromStructure = async (structure: FullStructure) => {
+  const constructReviewInput: ConstructReviewInput = (structure) => {
     const elements = Object.values(structure?.elementsById || {})
-    // Exclude not assigned, not visible and missing responses
-    const reviewableElements = elements.filter(
-      (element) => element?.isAssigned && element?.element.isVisible && element.response?.id
+    // Only create review for elements that are pendingReview
+    const reviewableElements = elements.filter(({ isPendingReview }) => isPendingReview)
+
+    const reviewResponseCreate = reviewableElements.map(
+      ({ latestPreviousLevelReviewResponse, response, reviewQuestionAssignmentId }) => {
+        // link to applicaiton response or review response based on review level
+        const applicationResponseId = assignment.level > 1 ? undefined : response?.id
+        const reviewResponseLinkId =
+          assignment.level === 1 ? undefined : latestPreviousLevelReviewResponse?.id
+        return {
+          applicationResponseId,
+          reviewResponseLinkId,
+          reviewQuestionAssignmentId,
+        }
+      }
     )
 
-    const applicationResponses = reviewableElements.map((element) => ({
-      applicationResponseId: element.response?.id || 0,
-      reviewQuestionAssignmentId: element.assignmentId,
-    }))
-    const result = await createReviewMutation({
+    return {
+      trigger: Trigger.OnReviewCreate,
+      reviewAssignmentId: assignment.id,
+      reviewResponsesUsingId: {
+        create: reviewResponseCreate,
+      },
+      // create new empty decision
+      reviewDecisionsUsingId: {
+        create: [{ decision: Decision.NoDecision }],
+      },
```

Comment:
Also another one that would be preferable set as a compulsory part of this `mutation` in my opinion.

Reply:
hmm, i see your point, was wondering what the difference is between setting it in mutation or here ? is it to protect someone using that mutation outside of this hook ?  (can we say only usable with this hook in the query comment, would that be sufficient ?

I would have to change to another type of mutation to set it explicitly in mutation, i was tying to avoid `ReviewResponseReviewIdFkeyReviewResponseCreateInput` etc.. which would need to be used

Reply:
As per comment above, I would need to re-write it, still thought quite interested to know why you inclined to have it set in mutation vs here (would this be enough, a comment: 173085881e1c39f7c16e5e960a1f4424cfa00067)

Reply:
Yes. That's fine, my worrying was to prevent bugs happening from changing this triggers. In this case it's pretty obvious, but some other cases (other tables that uses triggers) it might be not that obvious.